### PR TITLE
ILogger integration - part 1

### PR DESCRIPTION
**Pull request by reyang**
The 1st PR on ILogger integration.

Will explore the following things in subsequent PRs:
* Processor/Sampler.
* Exporter interface.
* Integration with trace id and span id.
* Integration with context/baggage.

## Changes

* Added basic skeleton - logger, logger provider, logger options.
* Added a tutorial project.

For significant contributions please make sure you have completed the following items:

* [x] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry/Logs/OpenTelemetryLoggerProvider.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                return this.scopeProvider;
+            }
+        }
+
+        void ISupportExternalScope.SetScopeProvider(IExternalScopeProvider scopeProvider)
+        {
+            this.scopeProvider = scopeProvider;
+        }
+
+        public ILogger CreateLogger(string name)
```
Comment (CodeBlanch) [inline src/OpenTelemetry/Logs/OpenTelemetryLoggerProvider.cs:55]: On the ILoggerProvider interface param is actually `categoryName` I think (going from memory), might want to rename here and on OpenTelemetryLogger ctor?

File: src/OpenTelemetry/Logs/OpenTelemetryLoggerProvider.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                    this.scopeProvider = new LoggerExternalScopeProvider();
+                }
+
+                return this.scopeProvider;
+            }
+        }
+
+        void ISupportExternalScope.SetScopeProvider(IExternalScopeProvider scopeProvider)
+        {
+            this.scopeProvider = scopeProvider;
```
Comment (CodeBlanch) [inline src/OpenTelemetry/Logs/OpenTelemetryLoggerProvider.cs:52]: I think we're missing some logic here. We need to loop through all the previously created `OpenTelemetryLogger`s and set their ScopeProvider to this one.

File: src/OpenTelemetry/Logs/OpenTelemetryLoggerProvider.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        }
+
+        public ILogger CreateLogger(string name)
+        {
+            return this.loggers.GetOrAdd(name, name => new OpenTelemetryLogger(name, this.options.CurrentValue));
```
Comment (CodeBlanch) [inline src/OpenTelemetry/Logs/OpenTelemetryLoggerProvider.cs:57]: Can we switch to calling TryGetValue/TryAdd explicitly? The bummer about the GetOrAdd interface is the lambda allocates a delegate on each call, used or not.

Comment (reyang) [reply src/OpenTelemetry/Logs/OpenTelemetryLoggerProvider.cs:57]: That's a great suggestion!

File: src/OpenTelemetry/Logs/OpenTelemetryLoggerProvider.cs (hunk 2 above)
Comment (reyang) [reply src/OpenTelemetry/Logs/OpenTelemetryLoggerProvider.cs:52]: Yes, this is a TODO.

**Review by CodeBlanch — APPROVED** [review]
LGTM

File: src/OpenTelemetry/Logs/OpenTelemetryLoggerProvider.cs (hunk 1 above)
Comment (reyang) [reply src/OpenTelemetry/Logs/OpenTelemetryLoggerProvider.cs:55]: Let me look into this, I was following the dotnet runtime code https://github.com/dotnet/runtime/blob/932cc5e7e38098c75aba0596b6375a0ddba8a0d8/src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerProvider.cs#L124.

Comment (reyang) [reply src/OpenTelemetry/Logs/OpenTelemetryLoggerProvider.cs]: You're right @CodeBlanch, it is indeed categoryName according to the [official doc](https://docs.microsoft.com/dotnet/api/microsoft.extensions.logging.iloggerprovider.createlogger#parameters). Thanks for the catch! 👍

File: src/OpenTelemetry/Logs/OpenTelemetryLoggerProvider.cs (hunk 2 above)
Comment (reyang) [reply src/OpenTelemetry/Logs/OpenTelemetryLoggerProvider.cs:54]: I've added a TODO comment.

**Review by cijothomas — APPROVED** [review]

**Pull request merged**